### DISCUSSION TITLE
failed Hot reload / rebuilt dev after 2nd time

### DISCUSSION THREAD

**Discussion by shaylavi** [gatsbyjs/gatsby#36865] [Help] [answered]
Hi there,

Exactly 1 year ago I wrote an awesome project with Gatsby and left it aside. Just this week I picked it up again and to my surprised 2 things happened -

1- My staticQuery image is no longer showing. It's just showing "Loading (StaticQuery)" instead of it.

2- I notice that if I change something in my code inside the layout component, **for the first save** it compiles it, rebuilds and shows the change on-screen successfully, but **from that point onwards**, it will detect the changes but will not show the changes anymore until I restart the whole Gatsby server. If I change a component outside the layout, it WILL save and show it, then if I restore the layout child component with the new changes, it still ignores it and shows the previous version.

I've been chasing this error for days now with no success - I reduced my site complexity to nearly nothing. created a new separate project to port my work into it piece by piece, I uninstalled all the "special" plugins to have only the basics.. nothing seems to fix this, yet I don't see any posts about this on google, which really surprises me. Someone suggested adding a flag in the config and it didn't solve it. I uninstalled Gatsby completely, cleaned the project, and reinstalled it all, I've stopped using yarn to have only npm in place... I'm just all out of ideas!

Here are some of the links I found and they did not help to fix this -

https://stackoverflow.com/questions/60954457/what-might-cause-gatsby-develop-to-get-stuck-while-building-development-bundl

https://github.com/gatsbyjs/gatsby/discussions/34741

https://github.com/gatsbyjs/gatsby/issues/34804

https://github.com/gatsbyjs/gatsby/issues/10836

I'm working on Windows 10 with the latest version.

Just a year ago, it all worked absolutely flawlessly! I changed nothing and it just happened!

Any suggestions will be appreciated.

Cheers,
Shay

**Comment by KyleAMathews** [comment]
Normally what I do here is remove node_modules & my lock file and then
reinstall. Did you try that with any luck? Next I'd try upgrading
everything to the latest version.
--
Kyle Mathews

Blog: *http://bricolage.io <http://bricolage.io>*
Twitter: http://twitter.com/kylemathews

**Reply by shaylavi** [reply]
Thank you, Kyle. I did try that and just retried it again. I did gatsby clean, deleted cache, public, lockfile and node_modules. npm install then gatsby develop - First save refreshes the changes in the browser, second save and beyond just ignores all changes although the console says it was rebuilt.

My code is using the latest version - 4.24.5.

This issue only applies to a sub-component inside layout. when I update other things outside of layout the data refreshes.

I'm really surprised I couldn't much info about this issue.
I think I'll try to recreate it from scratch and report it as a bug if I manage to do so

**Comment by KyleAMathews** [comment]
A reproduction would be very helpful to figure out the issue!

**Comment by shaylavi** [comment, marked as answer]
I created a new project and bit by bit tried really slowly to find the issue, but I managed to recreate the site without the problem and found some old conflicting dependency versions that may have caused it. I'll close this as I worked around the problem.
